### PR TITLE
Apply extra check on string passed in

### PR DESCRIPTION
**Pull request by wilsonge**
See: https://github.com/joomla/joomla-cms/commit/ff532960257e4a35c7218d2c779fe27da2741378

### CODE REVIEW

File: src/Path.php [hunk 1]
```diff
@@ -189,7 +189,7 @@ public static function check($path)
 	 */
 	public static function clean($path, $ds = DIRECTORY_SEPARATOR)
 	{
-		if (!is_string($path))
+		if (!is_string($path) && !empty($path))
```
Comment (mbabker) [inline src/Path.php]: If it's not a string and not empty, throw an exception?  Maybe I'm not thinking straight but something seems funny about that.

Comment (wilsonge) [inline src/Path.php]: So if you pass in an empty string then ```!empty($path)``` fails

Comment (mbabker) [inline src/Path.php]: That's what's getting me confused.  It almost seems like it's saying "if this is a string and it has characters throw an exception".

**Comment by mbabker** [comment]
I added this test case to `PathTest::getCleanData()`:

`'Empty string.' => array('', '/', JPATH_ROOT),`

Pre or post patch it doesn't seem to do change anything.  I can't make sense of what the change is aiming to solve.

**Comment by wilsonge** [comment]
It originates from this forum post http://forum.joomla.org/viewtopic.php?f=615&t=800374&p=3024730#p3024730

**Comment by mbabker** [comment]
Assortment of test cases I added to `PathTest::getCleanData()` to try and understand this.

```sh
'Empty array.' => array(array(), '/', JPATH_ROOT),

pre-patch
1) PathTest::testClean with data set "Empty array." (array(), '/', '/Users/mbabker/Sites/Joomla-F...system')
UnexpectedValueException: JPath::clean $path is not a string.

post-patch:
1) PathTest::testClean with data set "Empty array." (array(), '/', '/Users/mbabker/Sites/Joomla-F...system')
trim() expects parameter 1 to be string, array given
```

```sh
'Non-empty array.' => array(array('foo', 'bar'), '/', JPATH_ROOT),

pre- and post- patch
1) PathTest::testClean with data set "Non-empty array." (array('foo', 'bar'), '/', '/Users/mbabker/Sites/Joomla-F...system')
UnexpectedValueException: JPath::clean $path is not a string.
```

```
'Null string.' => array(null, '/', JPATH_ROOT),

pre-patch
1) PathTest::testClean with data set "Null string." (null, '/', '/Users/mbabker/Sites/Joomla-F...system')
UnexpectedValueException: JPath::clean $path is not a string.

post-patch
No failures
```

So something isn't quite right today with the environment from 2 years ago or the fix really didn't cover the intended cases.

**Comment by peter279k** [comment]
The `src/Path.php` looks like has the conflicts.

It should be fixed firstly.

File: src/Path.php [hunk 2]
```diff
@@ -215,7 +215,7 @@ public static function check($path)
 	 */
 	public static function clean($path, $ds = DIRECTORY_SEPARATOR)
 	{
-		if (!\is_string($path))
+		if (!\is_string($path) && !empty($path))
```
Comment (nibra) [inline src/Path.php:218]: Should be 
`if (!\is_string($path) || empty($path)) {...}`
to make sense.

On the other hand: Why not 

```php
if (!\is_string($path)) {...}

if (empty(trim($path))) {
    $path = JPATH_ROOT;
}
```
?

**Pull request closed without merging**